Frontend/MoptPaymentPayone/Controllers/Frontend/MoptAjaxPayone.php

```diff
@@ -925,6 +924,20 @@ protected function getOrderReferenceDetailsAction()
                 return;
             }
 
+            //check telephonenumber
+            if (Shopware()->Config()->get('requirePhoneField')){
+                $shipping_telephonenumber = $responseAddress['shipping_telephonenumber'];
+                if (strlen($shipping_telephonenumber) < 5 || !is_numeric($shipping_telephonenumber)){
```

Comment:
`is_numeric()` might be too strict here, as it denies phone numers like "01234/567890" or "301-555-1234", which are valid in some locales.